python/ray/data/_internal/execution/streaming_executor_state.py

```diff
@@ -266,6 +266,9 @@ def _execution_allowed(
         object_store_memory=global_usage.object_store_memory,
     )
     inc = op.incremental_resource_usage()
+    # If there are no resource demands, always allow execution.
```

Comment:
Since this bypasses the check of current object memory usage v.s. limit, this can create OOM if many tasks are launched and producing a lot objects?

Reply:
Potentially, yes (well, it would cause spilling at least). A safer fix may be https://github.com/ray-project/ray/pull/32673 (WIP).